websocket.go

```diff
@@ -193,15 +193,30 @@ func (t wsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 		s := t.service
 		var reply []byte
+		var replyChan chan []byte
 		path := strings.TrimPrefix(r.URL.Path, "/"+t.serviceName+"/")
 		log.Lvlf2("ws request from %s: %s/%s", r.RemoteAddr, t.serviceName, path)
-		reply, err = s.ProcessClientRequest(r, path, buf)
+		reply, replyChan, err = s.ProcessClientRequest(r, path, buf)
 		if err == nil {
-			tx += len(reply)
-			err := ws.WriteMessage(mt, reply)
-			if err != nil {
-				log.Error(err)
-				return
+			if replyChan == nil {
+				tx += len(reply)
+				err := ws.WriteMessage(mt, reply)
+				if err != nil {
+					log.Error(err)
+					return
+				}
+			} else {
+				go func() {
```

Comment:
Whacky. So if a client issues 2 requests on one websocket channel, and both have a streaming return, then the results from the two requests will arrive on the websocket, intermingled. And probably the call to ws.WriteMessage is racy.

In other words: I'm not sure about that `go`. :)

I was expecting that a ws connection used for streaming would be dedicated to delivering results until closed. (Not loop back to read again.)